### ISSUE TITLE
Allow NatSpec tag `@inheritdoc` to be used with contracts

### ISSUE THREAD

**Issue by PaulRBerg** [ethereum/solidity#14045]
## Motivation

Many Ethereum-based protocols, such Uniswap, choose to document their functions only in their interface files, [applying the `@inheritdoc` NatSpec tag in the implementation functions](https://twitter.com/PaulRBerg/status/1536002060818305024).

It would be nice to be able to do the same thing for contracts themselves, i.e. keep the documentation in the interface, and apply  `@inheritdoc` in the implementation. This would help with:

1. Avoid duplication.
2. Automate documentation generation with tools like `forge doc`

Applying this tag to contracts with the latest version of Solidity (v0.8.19 at the time of posting this) produces the following error:

> Documentation tag @inheritdoc not valid for contracts.

## Specification

```solidity
/// @notice This is my interface
interface MyInterface {}

/// @inheritdoc MyInterface
contract MyContract is MyInterface {}
```

## Backwards Compatibility

Adding support for this should be entirely backward-compatible.

**Comment by Coderayush13** [comment]
hey can you assign this issue to me

**Comment by nikola-matic** [comment]
> hey can you assign this issue to me

Hi @Coderayush13, this is not a good first issue, *and* we haven't decided whether we are going to implement it yet, so for the time being, no.

@PaulRBerg sounds reasonable to me. @ekpyron @cameel what do you think?

**Comment by CJ42** [comment]
I agree with @PaulRBerg that this would be a good addition to avoid duplication.

The same problem exists with `constructor`, where the tag `@inheritdoc` cannot be used.

For instance see this code snippet

<img width="498" alt="image" src="https://github.com/ethereum/solidity/assets/31145285/8f065ec4-ef5a-43a5-bdc7-732c20fff884">
